shared/constants/teams.js

```diff
@@ -229,8 +229,13 @@ const getTeamPublicitySettings = (state: TypedState, teamname: Types.Teamname):
 const getTeamInvites = (state: TypedState, teamname: Types.Teamname): I.Set<Types.InviteInfo> =>
   state.teams.getIn(['teamNameToInvites', teamname], I.Set())
 
+// Note that for isInTeam and isInSomeTeam, we don't use 'teamnames',
+// since that may contain subteams you're not a member of.
+
 const isInTeam = (state: TypedState, teamname: Types.Teamname): boolean =>
-  state.teams.hasIn(['teamnames', teamname])
+  state.teams.hasIn(['teamNameToRole', teamname])
```

Comment:
Subteams you're not in are still in the map, but with a role of `none`

Reply:
Fixed typing, per our discussion, and fixed this